include/kodlab_mjbots_sdk/lcm_message_handler.h

```diff
@@ -0,0 +1,74 @@
+/**
+ * @file lcm_message_handler.h
+ * @author Ethan Musser (emusser@seas.upenn.edu)
+ * @brief Provides LCM message handler object to be derived or composed into
+ * classes requiring LCM input via the `LcmSubscriber` object.
+ * @date 8/6/22
+ * 
+ * @copyright Copyright 2022 The Trustees of the University of Pennsylvania. All
+ * rights reserved.
+ * 
+ */
+
+#pragma once
+
+#include <string>
+#include <mutex>
+#include "lcm/lcm-cpp.hpp"
+#include "real_time_tools/thread.hpp"
+
+namespace kodlab {
+
+/**
+ * @brief Handler class for incoming LCM messages.
+ * @details This class is intended to be derived from or composed into objects
+ * requiring LCM input.  This class and its children should be compatible with
+ * the `LcmSubscriber` object, and can implemented with the
+ * `LcmSubscriber::AddSubscription` method in particular.
+ * @tparam Message LCM message type
+ */
+template<class Message>
+struct LcmMessageHandler {
+  /**
+   * @brief Flag indicating that a new message is available.
+   */
+  bool new_message = false;
+
+  /**
+   * @brief Data for the most-recently received message.
+   */
+  Message data;
```

Comment:
data should probably be a private variable and protected by mutexes for reading. Currently this happens externally via "SafeProcessInput" so that the functions dont take any extra parameters. Mentioned in https://github.com/KodlabPenn/kodlab_mjbots_sdk/pull/47#r939812149

Reply:
Resolved in ba8dcf1ad7d58d23fd9b58d79243d8c08754743c